docker-compose.yml

```diff
@@ -7,19 +7,71 @@ services:
     ports:
       - "8000:8000"
     volumes:
-      - ./thr:/thr
-    command: sh -c "python manage.py runserver 0.0.0.0:8000"
+      - .:/thr
+    command: >
+      sh -c "python manage.py makemigrations &&
+             python manage.py migrate &&
+             # Build elasticsearch index
+             #python manage.py search_index --rebuild -f &&
+             # Create the super user automatically using environment variables below
+             # || true; is added to ignore the error in case the user exists already
+             # It works only with Django 3+
+             #python manage.py createsuperuser --noinput || true;
+             python manage.py runserver 0.0.0.0:8000"
     environment:
       - DEBUG=1
+      - DJANGO_SUPERUSER_USERNAME=admin
+      - DJANGO_SUPERUSER_EMAIL=admin@mail.com
+      - DJANGO_SUPERUSER_PASSWORD=password
     depends_on:
-      - mysql_db
+      - mysql
+      - elasticsearch
+    networks:
+      - dbnet
+      - stacknet
 
-  mysql_db:
-    image: mysql:5.7
+  mysql:
+    container_name: mysql
+    image: mysql:$MYSQL_VERSION
     ports:
       - "3306:3306"
     environment:
       MYSQL_DATABASE: "thr_db"
       MYSQL_USER: "thr_dev"
       MYSQL_PASSWORD: "password"
-      MYSQL_ROOT_PASSWORD: "password"
\ No newline at end of file
+      MYSQL_ROOT_PASSWORD: "password"
+    networks:
+      - dbnet
+
+  elasticsearch:
+    container_name: elasticsearch
+    image: docker.elastic.co/elasticsearch/elasticsearch:$ELASTIC_VERSION
+    ports:
+      - "9200:9200"
+      - "9300:9300"
+    environment:
+      # https://discuss.elastic.co/t/problems-with-access-to-elasticsearch-form-outside-machine/172450
+      - discovery.type=single-node
+      - "ES_JAVA_OPTS=-Xms2g -Xmx2g"
```

Comment:
This won't be enough depending on your index mapping. ES likes to bloat heavily when people run certain queries and I found it was going into garbage collection hell on 2GB. It's that or a healthy reboot cycle.

Reply:
How much memory do you think is reasonable?
I disabled _dynamic mapping_, thankfully this will decrease the memory usage